### ISSUE TITLE
Code marked as unreachable if type hints don't include None but variable is intialised as None

### ISSUE THREAD

**Issue by lorddevereux** [microsoft/pyright#8493]
I have seen a change in behaviour recently which is causing some issues where code blocks are (in my opinion) wrongly marked as unreachable. From some digging, this behaviour changed in 1.1.369 from the following commit: https://github.com/microsoft/pyright/commit/b841e110f31dd552d1da9eaabb58c100881f0bd5 

A minimal example:

```
def myfunc(a: int = None):
    if a is None:
         print("this line marked as unreachable")
```

Before 1.1.369, this line would not be marked as unreachable.

I have the following option set to "none" in diagnosticSeverityOverrides, which I believe (semantically) should prevent this from happening by implicitly assuming that a is **int|None** without the type hint needing to specify. However, it doesn't seem to work.

> strictParameterNoneValue [boolean]: PEP 484 indicates that when a function parameter is assigned a default value of None, its type should implicitly be Optional even if the explicit type is not. 

I'm aware that best practice has moved to requiring explicitly specifing None, however I think this setting should be respected by the unreachable code checker. There are some large codebases which rely on this implicit behaviour and so it is quite annoying for large chunks to now be [wrongly] marked as "unreachable".

**Comment by erictraut** [comment]
The `strictParameterNoneValue ` is a boolean switch. It is not a diagnostic rule, so it doesn't support diagnostic severity controls, and it isn't supported in the `diagnosticSeverityOverrides` setting. If you want to disable it, you need to do so in your pyright configuration file (either `pyrightconfig.json` or `pyproject.toml` file).

For more information, refer to the [configuration documentation](https://microsoft.github.io/pyright/#/configuration).

**Comment by lorddevereux** [comment]
Ok, thanks, that's working as expected with a local config file. For anyone who finds this, it's not obvious but having dug through the documentation again it does in fact say
> "reportXXX" rules in the type check diagnostics settings in [configuration](https://microsoft.github.io/pyright/#/configuration?id=type-check-diagnostics-settings) are supported. Use the rule name as a key and one of "error," "warning," "information," "true," "false," or "none" as value.

It's annoying that there's no way to apply the other configurations globally, but that's a separate issue.
